graphql_compiler/tests/test_input_data.py

```diff
@@ -680,6 +680,41 @@ def simple_recurse():
         type_equivalence_hints=None)
 
 
+def double_recurse():
+    graphql_input = '''{
+        Animal @filter(op_name: "name_or_alias", value: ["$animal_name_or_alias"]) {
+            name @output(out_name: "animal_name")
+            out_Animal_ImportantEvent @optional {
```

Comment:
Is the `@optional` required to trigger the bug? If the bug happens without it, let's remove it -- we're aiming for the minimal possible repro.